lib/MusicBrainz/Server/Data/Role/QueryToList.pm

```diff
@@ -16,13 +16,19 @@ sub query_to_list {
 }
 
 sub query_to_list_limited {
-    my ($self, $query, $args, $limit, $offset, $builder) = @_;
+    my ($self, $query, $args, $limit, $offset, $builder, $dollar_placeholders) = @_;
 
     $builder //= $self->can('_new_from_row');
 
     my @args = @$args;
+    my $count = @args;
+
     if (defined $offset) {
-        $query .= " OFFSET ?";
+        if ($dollar_placeholders) {
+            $query .= ' OFFSET $' . (++$count);
```

Comment:
Also, what's the difference between using `?` placeholders and `$<value>`? I don't understand why there need to be two ways to pass these arguments.